### ISSUE TITLE
Oracle 12c OCIError: Message 1359481369 not found

### ISSUE THREAD

**Issue by areman** [kubo/ruby-oci8#87]
Hello guys,

I'm using ruby oci with an oracle 12c. Sometime a error like ...
OCIError: ORA-1359481369: Message 1359481369 not found;  product=RDBMS; facility=ORA: SELECT count(*) as i_itemqty FROM ...
is raised. I'm not sure whats the problem is. Can anybody give me a hit how to fix it ?

Best Regards,
André

**Comment by kubo** [comment]
Could you post more detailed information to reproduce the issue?
* Oracle versions both client and server
* OS names and versions both client and server
* ruby version
* ruby-oci8 version
* test script which produces the issue

**Comment by kubo** [comment]
Close this issue because nobody can fix a bug without information.
If you can reproduce it, please report it again.

**Comment by dbernheisel** [comment]
@areman did you ever figure it out? I'm running into a similar issue with ruby 1.9.3p551, ruby-oci8 2.2.2, 12.1 xe on a redhat server. For me, the issue comes up when querying against a database view that has a `program_id` column, but it's really inconsistent. In one moment, `MyDbViewModel.where(program_id: 123)` didn't work, but then `MyDbViewModel.where('program_id = ?', 123)` did work; but then both styles immediately started working.

**Comment by dbernheisel** [comment]
I was able to confirm that behavior is only with Oracle 12c and not with Oracle 11g.

**Comment by dbernheisel** [comment]
With some additional testing using `OCI8` instead of going through the ActiveRecord adapter, I could see that it's not an oci8 issue. I'm using [release14](https://github.com/rsim/oracle-enhanced/tree/release14) of the [oracle-enhanced](https://github.com/rsim/oracle-enhanced) adapter.

**Comment by areman** [comment]
strange. I can't belive. you think it's a oracle-enhanced problem? Maybe the oracle-enhanced-adapter sets some special session parameters.

**Comment by dbernheisel** [comment]
Dropping straight to ruby-oci8 didn't have a problem with the query. Using the oracle-enhanced adapter introduces the issue.

Looks like the oracle-enhanced-adapter team fixed a bug in their adapter in a later version. I only encountered it when using Release14 (latest for Ruby 1.9.3). I spiked a new project using a later release of the adapter and the same database and didn't encounter the issue.

The problem seems rooted in how the adapter handles floats versus integers.

So, what I guess this means is that I backport fixes to Release14 of the adapter, or I convince the client to upgrade my app's Ruby/Rails versions so I can upgrade the adapter.
